src/engine/StripColumns.cpp

```diff
@@ -0,0 +1,132 @@
+std::string StripColumns::getCacheKeyImpl() const {
+  return absl::StrCat("StripColumns(", absl::StrJoin(subset_, ","), " - ",
+                      child_->getCacheKey(), ")");
+}
+
+// _____________________________________________________________________________
+std::string StripColumns::getDescriptor() const { return "Strip Columns"; }
+
+// _____________________________________________________________________________
+size_t StripColumns::getResultWidth() const { return subset_.size(); }
+
+// _____________________________________________________________________________
+size_t StripColumns::getCostEstimate() { return child_->getCostEstimate(); }
+
+// _____________________________________________________________________________
+uint64_t StripColumns::getSizeEstimateBeforeLimit() {
+  return child_->getSizeEstimate();
+}
+
+// _____________________________________________________________________________
+float StripColumns::getMultiplicity(size_t col) {
+  return col < subset_.size() ? child_->getMultiplicity(subset_.at(col)) : 1.0f;
+}
+
+// _____________________________________________________________________________
+bool StripColumns::knownEmptyResult() { return child_->knownEmptyResult(); }
+
+// _____________________________________________________________________________
+std::unique_ptr<Operation> StripColumns::cloneImpl() const {
+  return std::make_unique<StripColumns>(getExecutionContext(), child_->clone(),
+                                        subset_, varToCol_);
+}
+
+// _____________________________________________________________________________
+std::vector<ColumnIndex> StripColumns::resultSortedOn() const {
+  std::vector<ColumnIndex> sortedOn;
+  const auto& fromChild = child_->resultSortedOn();
+  // Find the largest prefix of `fromChild` that is part of the `subset` and
+  // return it with the columns mapped accordingly. In particular, if the child
+  // is ordered by three variables
+  // `?a ?b ?c`, but only `?a` and `?c` are preserved by this `StripColumns`
+  // operation, then the result of `*this` is only sorted by `?a`.
+  for (auto col : fromChild) {
+    auto it = ql::ranges::find(subset_, col);
+    if (it == subset_.end()) {
+      break;
+    }
+    // the position in the subset is the column index wrt `*this`.
+    sortedOn.push_back(it - subset_.begin());
+  }
+  return sortedOn;
+}
+
+// _____________________________________________________________________________
+Result StripColumns::computeResult(bool requestLaziness) {
+  auto res = child_->getResult(requestLaziness);
+  if (res->isFullyMaterialized()) {
+    // TODO<joka921> This case currently is inefficient, we should really
+    // implement moving the tables from materialized results that are too big
+    // for the cache or having a `shared_ptr<IdTable+SubsetView>` type in the
+    // result.
+    auto table = res->idTable().clone();
+    table.setColumnSubset(subset_);
```

Comment:
I believe we have an overload of `clone()` that just clones a specific set of columns.